course5/homework11/homework11.cc

```diff
@@ -0,0 +1,178 @@
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/4                  2 ns          2 ns  310438844
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/16                 6 ns          6 ns  118006811
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/64                81 ns         81 ns    8594129
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/256              367 ns        367 ns    1890390
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/1024            1091 ns       1091 ns     638900
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/4096           14110 ns      14110 ns      49528
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/16384          34955 ns      34956 ns      19839
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/65536         367182 ns     367186 ns       1881
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/262144        956106 ns     956115 ns        760
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/1048576     78667157 ns   78667654 ns          9
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/4194304    435734658 ns  435739838 ns          2
+//    container_iterate_ordered<std::deque<std::int64_t>, random_read>/16777216   938941599 ns  938953250 ns          1
+
+// Plotting the results:
+// $ ./homework11 --benchmark_repetitions=10 --benchmark_report_aggregates_only=true --benchmark_format=csv --benchmark_out=read_benchmark.csv
+// $ grep "vector.*linear_read.*mean" read_benchmark.csv | sed 's/"[^/]*\/\([0-9]\+\)_mean"/\1/' > vector_linear.csv
+// $ grep "vector.*random_read.*mean" read_benchmark.csv | sed 's/"[^/]*\/\([0-9]\+\)_mean"/\1/' > vector_random.csv
+// $ grep "deque.*linear_read.*mean" read_benchmark.csv | sed 's/"[^/]*\/\([0-9]\+\)_mean"/\1/' > deque_linear.csv
+// $ grep "deque.*random_read.*mean" read_benchmark.csv | sed 's/"[^/]*\/\([0-9]\+\)_mean"/\1/' > deque_random.csv
+// $ gnuplot
+// gnuplot> set datafile separator ","
+// gnuplot> plot 'vector_linear.csv' using 1:3 title "vector linear" with lines, 'vector_random.csv' using 1:3 title "vector random" with lines, 'deque_linear.csv' using 1:3 title "deque linear" with lines, 'deque_random.csv' using 1:3 title "deque random" with lines
+
+enum ReadMode { linear_read, random_read};
+
+template <class Container, ReadMode order>
+static void container_iterate_ordered(benchmark::State& state)
+{
+    Container c;
+    for (std::int64_t i=1; i<=state.range(0); ++i)
+        c.push_back(i);
+    if (order == random_read)
+    {
+        std::shuffle(std::begin(c), std::end(c), std::default_random_engine(seed));
+    }
+
+    while(state.KeepRunning())
+    {
+        size_t pos = 0;
+        while (pos<c.size())
```

Comment:
Does this correctly control the number of iterations?
if it happens so ```pos[0] = size```, then there will be only one iteration of the loop.